netmiko/mikrotik/mikrotik_ssh.py

```diff
@@ -117,3 +120,154 @@ class MikrotikSwitchOsSSH(MikrotikBase):
     """Mikrotik SwitchOS SSH driver."""
 
     pass
+
+
+class MikrotikRouterOsFileTransfer(BaseFileTransfer):
+    """Mikrotik Router Os File Transfer driver."""
+
+    def __init__(
+        self,
+        ssh_conn: BaseConnection,
+        source_file: str,
+        dest_file: str,
+        file_system: Optional[str] = "flash",
+        direction: str = "put",
+        socket_timeout: float = 10.0,
+        progress: Optional[Callable[..., Any]] = None,
+        progress4: Optional[Callable[..., Any]] = None,
+        hash_supported: bool = False,
+    ) -> None:
+        super().__init__(
+            ssh_conn=ssh_conn,
+            source_file=source_file,
+            dest_file=dest_file,
+            file_system=file_system,
+            direction=direction,
+            socket_timeout=socket_timeout,
+            progress=progress,
+            progress4=progress4,
+            hash_supported=hash_supported,
+        )
+
+    def check_file_exists(self, remote_cmd: str = "") -> bool:
+        """Check if the dest_file already exists on the file system."""
+        if self.direction == "put":
+            if not remote_cmd:
+                remote_cmd = f'/file print detail where name="{self.file_system}/{self.dest_file}"'
+            remote_out = self.ssh_ctl_chan._send_command_timing_str(remote_cmd)
+            # Output will look like
+            # 0 name="flash/test9.txt" type=".txt file" size=19 creation-time=jun...
+            # fail case will be blank line (all whitespace)
+            if (
+                "size" in remote_out
+                and f"{self.file_system}/{self.dest_file}" in remote_out
+            ):
+                return True
+            elif not remote_out.strip():
+                return False
+            raise ValueError("Unexpected output from check_file_exists")
```

Comment:
And now that I think about it, this is why I had that complicated command

```
                remote_cmd = (
                    '{local f FOUND;:if ([:len [/file find name="'
                    f'{self.file_system}/{self.dest_file}"]] > 0) '
                    'do={{:put "YES_$f"}} else={{:put "NOT_$f"}}}'
                )
```

In that way it will not find the "YES_FOUND" or "NOT_FOUND" in the prompt command because I pass by a local variable

Reply:
Okay, so it sounds like we are gated by fixing #2797 first then...